tests/testthat/test-read.R

```diff
@@ -74,6 +75,61 @@ test_that("Reading H5AD works with native reader", {
     expect_identical(colnames(colData(sce)), "cell_type")
 })
 
+test_that("Reading v0.8 H5AD works with native reader", {
+    sce_py <- readH5AD(file_v08)
+    sce_r <- readH5AD(file_v08, reader='R')
+
+    expect_identical(rownames(sce_py), rownames(sce_r))
+    expect_identical(colnames(sce_py), colnames(sce_r))
+
+    # rowData is just an empty DataFrame here
+    expect_identical(rowData(sce_py), rowData(sce_r))
+
+    # colData colnames not in same order, but check the sets are the same
+    expect_true(setequal(colnames(colData(sce_py)), colnames(colData(sce_r))))
```

Comment:
Would it be possible/easy to maintain the order? I'm pretty sure the Python reader keeps it so it should be in the fine somewhere (although not sure I have properly checked this).

Reply:
There's a `"column-order"` attribute on dataframes for this (https://anndata.readthedocs.io/en/latest/fileformat-prose.html#dataframes)

Reply:
Updated to use the `column-order` attribute